workspaces-tests/src/bin/simple_multisig.rs

```diff
@@ -0,0 +1,126 @@
+#![allow(missing_docs)]
+
+// Ignore
+pub fn main() {}
+
+use std::fmt::Display;
+
+use near_contract_tools::{
+    approval::{
+        self,
+        simple_multisig::{AccountAuthorizer, ApprovalState, Configuration},
+        ApprovalManager,
+    },
+    rbac::Rbac,
+    slot::Slot,
+    Rbac,
+};
+use near_sdk::{
+    borsh::{self, BorshDeserialize, BorshSerialize},
+    env, near_bindgen, BorshStorageKey, PanicOnDefault,
+};
+
+#[derive(BorshSerialize, BorshStorageKey)]
+enum StorageKey {
+    SimpleMultisig,
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+enum MyAction {
+    SayHello,
+    SayGoodbye,
+}
+
+impl approval::Action for MyAction {
+    type Output = &'static str;
+
+    fn execute(self) -> Self::Output {
+        match self {
+            Self::SayHello => "hello",
+            Self::SayGoodbye => "goodbye",
+        }
+    }
+}
+
+#[derive(BorshSerialize, BorshStorageKey)]
+enum Role {
+    Multisig,
+}
+
+#[derive(PanicOnDefault, BorshSerialize, BorshDeserialize, Rbac)]
+#[rbac(roles = "Role")]
+#[near_bindgen]
+pub struct Contract {}
+
+// This single function implementation completely implements simple multisig on
+// the contract
+impl ApprovalManager<MyAction, ApprovalState, Configuration<Self>> for Contract {
+    fn root() -> Slot<()> {
+        Slot::new(StorageKey::SimpleMultisig)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum ApproverError {
+    UnauthorizedAccount,
+}
+
+impl Display for ApproverError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Unauthorized account")
+    }
+}
+
+// We don't have to check env::predecessor_account_id or anything like that
+// SimpleMultisig handles it all for us
+impl AccountAuthorizer for Contract {
+    fn is_account_authorized(account_id: &near_sdk::AccountId) -> bool {
+        Contract::has_role(account_id, &Role::Multisig)
+    }
+}
+
+#[near_bindgen]
+impl Contract {
+    const APPROVAL_THRESHOLD: u8 = 2;
+    const VALIDITY_PERIOD: u64 = 1000000 * 1000 * 60 * 60 * 24 * 7;
```

Comment:
@encody Did the linter not work here (and at https://github.com/NEARFoundation/near-contract-tools/pull/39/files#diff-5f3aae4e99977bf4035d5b96decfba59a2951d5758ed867317ae995b4ff62696R204 and probably elsewhere too)?

Reply:
The best linter rule I could find only supports inconsistent digit groupings, it doesn't enforce their use.